[Task_1] Создан проект ASP.NET Core 5.0 Web API

SHILY-PROJECT opened the pull request.

SHILY-PROJECT committed:
-Создан проект ASP.NET Core 5.0 Web API

SHILY-PROJECT committed:
-Фикс типа проекта...))

SHILY-PROJECT committed:
-Созданы классы "Human" и "Book"
-Создан класс "DataStore" со списками для классов выше

SHILY-PROJECT committed:
-Реализован контроллер, который отвечает за человека

SHILY-PROJECT committed:
-Мелкий рефакторинг

SHILY-PROJECT committed:
-Реализован контроллер отвечающий за книгу
-Мелкий рефакторинг

SHILY-PROJECT committed:
-Создан класс "LibraryCard" для хранения данных людей, котоые взяли книгу
-Создан контроллер отвечающий за получение книги (+метод отвечающий за взятие книги читателем)
-Создан пустой статичный список, отвечающий за хранение этих сущностей

SHILY-PROJECT committed:
-Микро рефакторинг

SHILY-PROJECT committed:
-Ренейм и мелкий рефакторинг

SHILY-PROJECT committed:
-Добавлена сортировка при получении списка книг
-Фикс идентификатора при добавлении нового человека

Skreet2k commented on SimbirSoftWorkshop.API/Controllers/BookController.cs:
```diff
@@ -0,0 +1,119 @@
+        /// </summary>
+        [HttpGet("listAllBooks")]
+        public IActionResult GetBooks(SortBookOptionsEnum sortOprion)
+            => Ok(sortOprion switch
```
Имя параметра кривое и в идеале для всех параметров указывать откуда оно. В твоем случае [FromQuery]

Skreet2k commented on SimbirSoftWorkshop.API/Controllers/BookController.cs:
```diff
@@ -0,0 +1,119 @@
+        [HttpPost("addNewBookToList")]
+        public IActionResult PostAddBook(string title, string genre, string nameAuthor, string surnameAuthor)
+        {
+            var variables = new List<(string nameVar, string valueVar)>
```
Вот это ты завернул. Для этого есть простой механизм - [Required]

Skreet2k commented on SimbirSoftWorkshop.API/Controllers/BookController.cs:
```diff
@@ -0,0 +1,119 @@
+        /// <param name="authorName">Имя автора</param>
+        /// <param name="authorSurname">Фамилия автора</param>
+        /// <returns></returns>
+        [HttpDelete("deleteBookFromList")]
```
У тебя уже есть глагол в методе Delete. Не стоит в URL добавлять стандартные глаголы HTTP методов

Skreet2k commented on SimbirSoftWorkshop.API/Controllers/HumanController.cs:
```diff
@@ -0,0 +1,116 @@
+        /// </summary>
+        /// <param name="searchQuery">Поисковый запрос (фраза).</param>
+        [HttpGet("listHumansBySearchQuery")]
+        public IActionResult GetHumansByLineQuery(string searchQuery)
```
Самое сложное - это придумавать названия переменных/классов/методов. Вернее GetHumansByQuery

Skreet2k replied:
[Required]
[MinLength(1)]

Skreet2k commented on SimbirSoftWorkshop.API/Controllers/WeatherForecastController.cs:
```diff
@@ -0,0 +1,38 @@
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class WeatherForecastController : ControllerBase
```
Зачем оставил?

Skreet2k commented on SimbirSoftWorkshop.API/Models/LibraryCard.cs:
```diff
@@ -0,0 +1,21 @@
+        {
+            Human = human;
+            Book = book;
+            BookReceiptDate = DateTime.Now;
```
В классе Offset, а при создании используешь DateTimeNow

Skreet2k commented on SimbirSoftWorkshop.API/Models/HumanDto.cs:
```diff
@@ -0,0 +1,18 @@
+        public string Name { get; set; }
+        public string Surname { get; set; }
+        public string Patronymic { get; set; }
+        public long HumanId { get; set; }
```
Правильнее наверное Id. Мы и так знаешь что модель Human

Skreet2k commented on SimbirSoftWorkshop.API/Controllers/LibraryCardController.cs:
```diff
@@ -0,0 +1,60 @@
+            DataStore.LibraryCards.Add(libraryCard);
+
+            if (!DataStore.LibraryCards.Contains(libraryCard))
+                return BadRequest("Не удалось добавить библиотечную карточку");
```
Не правильная логика. Ты должен сделать Add - и если нет ошибок, нет смысла проверять на валидность выполненной операции, делая запрос на наличие модели в списке.


Другое дело если бы мы использовал метод TryAdd.

Skreet2k reviewed (approved):
"Плюсы

Хорошо задокументированный код
Полностью выполненная работа

Минусы

Нет допов, есть оставшиеся от шаблона классы и контроллеры
Очень сложная бизнес логика, не использует Атрибуты DataAnnotation"

SHILY-PROJECT committed:
-Разделение сущностей
-Рефакторинг (ренейминг некоторых сущностей, методов и классов, фикс багов, добавление атрибутов валидации и т.д.)
-Чистка от мусора
